LanguageFeatures/Augmentation-libraries/augmenting_functions_A05_t02.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - The function augmentation specifies any default values.
+///
+/// @description Checks that it is a compile-time error if a function
+/// augmentation specifies any default values.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmenting_functions_A05_t02_lib.dart';
+
+void topLevelFunction1([int i]) {}
```

Comment:
Perhaps it would be safer to use a parameter type like `int?` (and everywhere, for this test) such that it is not an error to have this declaration in the first place?

It seems like we could easily misinterpret the situation when we have both "errors that we are testing" and "errors that just happen because we've written something that is wrong and isn't being tested".